### PR TITLE
registry config doc cleanup

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
[(3) Document using service DNS for the internal registry instead of IP address [registry]](https://trello.com/c/BRPwto2S/1309-3-document-using-service-dns-for-the-internal-registry-instead-of-ip-address-registry)

**Comment by bparees** [comment]
@dmage @legionus @miminar ptal.  This includes some additional clean up of the registry doc, but the main purpose is to document the equivalent to what the ansible installer does when you specify "openshift_push_via_dns".

Near as i can tell, all the ansible installer does under those circumstances is setup a `OPENSHIFT_DEFAULT_REGISTRY` env variable for the openshift master with a value of `docker-registry.default.svc:5000`

The certificate that is generated for the registry by ansible always includes the docker registry service hostname, but I have included a reference pointing users to how to generate the certificate w/ the appropriate hostnames as part of the instrucitons for configuring the registry hostname.

File: install_config/registry/extended_registry_configuration.adoc [hunk 1]
```diff
@@ -619,6 +658,20 @@ using the boolean environment variable
...
+environment variable `REGISTRY_OPENSHIFT_REQUESTS_READ_MAXRUNNING`.  Zero indicates no limit.
+<5> Maximum number of queued pull requests.  It can be overridden by the
+environment variable `REGISTRY_OPENSHIFT_REQUESTS_READ_MAXINQUEUE`.  Zero indicates no limit.
+<6> Maximum time a pull request can wait in the queue before being rejected.
+It can be overridden by the environment variable `REGISTRY_OPENSHIFT_REQUESTS_READ_MAXWAITINQUEUE`.
+Zero indicates no limit.
+<7> Maximum number of simultaneous push requests.  It can be overridden by the
+environment variable `REGISTRY_OPENSHIFT_REQUESTS_WRITE_MAXRUNNING`.  Zero indicates no limit.
+<8> Maximum number of queued push requests.  It can be overridden by the
+environment variable `REGISTRY_OPENSHIFT_REQUESTS_READ_MAXINQUEUE`.  Zero indicates no limit.
```
Comment (dmage) [inline install_config/registry/extended_registry_configuration.adoc]: s/READ/WRITE/

Comment (bparees) [reply install_config/registry/extended_registry_configuration.adoc]: thanks, fixed.

**Review by miminar — APPROVED** [review]
/lgtm

**Comment by bparees** [comment]
@openshift/team-documentation ptal/merge.

**Comment by bparees** [comment]
(because this doc includes information about rate-limiting, it does not apply prior to 3.7)

**Comment by bmcelvee** [comment]
LGTM. @bparees, since both files are in the Installation and Configuration section, they only apply to OCP. I'll go ahead and remove the online label, and cherry pick only to OCP 3.7.

**Comment by bmcelvee** [comment]
[rev_history]
|xref:../install_config/master_node_configuration.adoc#install-config-master-node-configuration[Master and Node Configuration]
|Updated the xref:../install_config/master_node_configuration.adoc#master-config-image-config[Image Configuration] and xref:../install_config/master_node_configuration.adoc#master-config-image-policy-config[Image Policy Configuration] tables.
%
|xref:../install_config/registry/extended_registry_configuration.adoc#install-config-registry-extended-configuration[Setting up the Registry -> Extended Registry Configuration]
|Added the xref:../install_config/registry/extended_registry_configuration.adoc#setting-the-registry-hostname[Setting the Registry Hostname] section, and updated the registry metrics collection example in the xref:../install_config/registry/extended_registry_configuration.adoc#docker-registry-configuration-reference-openshif[OpenShift] section.

**Pull request merged**